src/js/main.js

```diff
@@ -77,117 +88,122 @@ var SupportKit = Marionette.Object.extend({
+            };
+        }
+
         if (typeof options === 'object') {
             endpoint.appToken = options.appToken;
-            endpoint.jwt = options.jwt;
             if (options.serviceUrl) {
                 endpoint.rootUrl = options.serviceUrl;
             }
-        } else if (typeof options === 'string') {
-            endpoint.appToken = options;
         } else {
-            throw new Error('init method accepts an object or string');
+            return Promise.reject(new Error('init method accepts an object or string'));
         }
 
         if (!endpoint.appToken) {
-            throw new Error('init method requires an appToken');
+            return Promise.reject(new Error('init method requires an appToken'));
         }
 
-        this.deviceId = this.getDeviceId();
+        return this.login(options.userId, options.jwt);
+    },
 
-        api.call({
-                url: 'appboot',
-                method: 'POST',
-                data: {
-                    deviceId: this.deviceId,
-                    userId: options.userId,
-                    deviceInfo: {
-                        URL: document.location.host,
-                        userAgent: navigator.userAgent,
-                        referrer: document.referrer,
-                        browserLanguage: navigator.language,
-                        currentUrl: document.location.href,
-                        sdkVersion: this.VERSION,
-                        currentTitle: document.title,
-                        platform: 'web'
-                    }
-                }
-            })
-            .then(_(function(res) {
-                this.user = new AppUser({
-                    id: res.appUserId
-                });
+    login: function(userId, jwt) {
+        // clear unread for anonymous
+        if (!this.user.isNew() && userId && !this.user.get('userId')) {
+            // the previous user had no userId and it's switching to one with an id
+            this._chatController.clearUnread();
+        }
 
-                var EventCollection = Backbone.Collection.extend({
-                    url: urljoin('appusers', res.appUserId, 'event'),
-                    model: Event
-                });
+        this._cleanState();
+
+        var data = {
+            deviceId: this.getDeviceId(),
+            deviceInfo: {
+                URL: document.location.host,
+                userAgent: navigator.userAgent,
+                referrer: document.referrer,
+                browserLanguage: navigator.language,
+                currentUrl: document.location.href,
+                sdkVersion: this.VERSION,
+                currentTitle: document.title,
+                platform: 'web'
+            }
+        };
 
-                var RuleCollection = Backbone.Collection.extend({
-                    model: Rule
-                });
+        if (userId) {
+            data.userId = userId;
+            endpoint.userId = userId;
```

Comment:
As for reverting, I felt like the login action would always be logout first, then login. If login fails, log out still happened.